### DISCUSSION TITLE
How to add android SDK into koreader?

### DISCUSSION THREAD

**Discussion by mlouielu** [koreader/koreader#7273] [Q&A]
Hi there,

I'm only familiar to android studio, so I'm not sure how to add third-party SDK into koreader.

I'm using Samsung Tab, and I want to test about [S PEN remote SDK](https://developer.samsung.com/galaxy-spen-remote/overview.html), so I can control koreader with S PEN.

Are there any instruction or guide for this?

Thanks.

**Comment by Frenzie** [comment]
I think the https://github.com/koreader/android-luajit-launcher explains it reasonably well. Please let us know if anything's unclear.

**Reply by mlouielu** [reply]
Thanks for pointing out, I'll try to figure it out!

Thanks

**Reply by NiLuJe** [reply]
Something something license making it unredistributable or something, IIRC what @pazos said about that specific SDK in the past?

**Reply by pazos** [reply]
I'm not a lawyer :p 

Anyways https://github.com/koreader/android-luajit-launcher is MIT licensed

Being https://github.com/koreader/koreader AGPLv3 we can use other kind of more permissive licenses as long as the final application can be still legaly licensed as AGPLv3. We **cannot mix koreader with privative software**

Of course all of this is in the case the remote pen SDK is not Apache/MIT/BSD or similar, which I didn't check because samsung requires me to log into an account to download the SDK.

It is still possible to have a different flavor `samsung` that uses the SDK, with a dummy implementation on the `main flavor` and a common api exposed to koreader, and write koreader code that consumes that api. But users will need to build their own apk and cannot redistribute it
